src/pytorch_lightning/profilers/hpu.py

```diff
@@ -0,0 +1,173 @@
+
+        Args:
+            dirpath: Directory path for the ``filename``. If ``dirpath`` is ``None`` but ``filename`` is present, the
+                ``trainer.log_dir`` (from :class:`~pytorch_lightning.loggers.tensorboard.TensorBoardLogger`)
+                will be used.
+
+            filename: If present, filename where the profiler results will be saved instead of printing to stdout.
+                The ``.txt`` extension will be used automatically.
+
+            group_by_input_shapes: Include operator input shapes and group calls by shape.
+
+            export_to_chrome: Whether to export the sequence of profiled operators for Chrome.
+                It will generate a ``.json`` file which can be read by Chrome.
+
+            row_limit: Limit the number of rows in a table, ``-1`` is a special value that
+                removes the limit completely.
+
+            sort_by_key: Attribute used to sort entries. By default
+                they are printed in the same order as they were registered.
+                Valid keys include: ``cpu_time``, ``cpu_time_total``,
+                ``cpu_memory_usage``, ``self_cpu_memory_usage``, ``count``.
+
+            record_module_names: Whether to add module names while recording autograd operation.
+
+            profiler_kwargs: Keyword arguments for the PyTorch profiler. This depends on your PyTorch version
+
+        Raises:
+            MisconfigurationException:
+                If arg ``sort_by_key`` is not present in ``AVAILABLE_SORT_KEYS``.
+                If arg ``schedule`` is not a ``Callable``.
+                If arg ``schedule`` does not return a ``torch.profiler.ProfilerAction``.
+        """
+        super().__init__(
+            dirpath=dirpath,
+            filename=filename,
+            group_by_input_shapes=group_by_input_shapes,
+            export_to_chrome=export_to_chrome,
+            row_limit=row_limit,
+            sort_by_key=sort_by_key or f"{'hpu' if profiler_kwargs.get('use_hpu', False) else 'cpu'}_time_total",
+            record_module_names=record_module_names,
+        )
+
+        self.profiler: Optional[_PROFILER] = None
+        self.function_events: Optional["EventList"] = None
+        self._lightning_module: Optional["LightningModule"] = None  # set by ProfilerConnector
+        self._register: Optional[RegisterRecordFunction] = None
+        self._parent_profiler: Optional[_PROFILER] = None
+        self._recording_map: Dict[str, record_function] = {}
+        self._start_action_name: Optional[str] = None
+        self._schedule: Optional[ScheduleWrapper] = None
+        self._profiler_kwargs = profiler_kwargs
+
+        if _KINETO_AVAILABLE:
+            self._init_kineto(profiler_kwargs)
+
+        if self._sort_by_key not in self.AVAILABLE_SORT_KEYS:
+            raise MisconfigurationException(
+                f"Found sort_by_key: {self._sort_by_key}. Should be within {self.AVAILABLE_SORT_KEYS}. "
+            )
```

Comment:
Could we remove these lines because they are the exact same as in `PyTorchProfiler.__init__()` https://github.com/Lightning-AI/lightning/blob/9fc4ff3278ac360f2671549415f6b11689a263ed/src/pytorch_lightning/profilers/pytorch.py#L300-L317?

Reply:
@jerome-habana

Reply:
updated